appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java

```diff
@@ -474,11 +475,14 @@ public WeldApplicationContainer load(WeldContainer container, DeploymentContext
         String archiveName = !isSubArchive? appInfo.getName() : archive.getName();
         int suffixIdx = archiveName.lastIndexOf('-');
         if(isSubArchive && suffixIdx > 0) {
-            archiveName = archiveName.substring(0, suffixIdx);
-            if(!context.getArchiveHandler().getArchiveType().isEmpty()) {
-                archiveName = String.format("%s.%s", archiveName, context.getArchiveHandler().getArchiveType());
+            String versionStr = archiveName.substring(suffixIdx + 1, archiveName.length());
+            if(versionStr.matches("^[0-9]+\\..*")) {
```

Comment:
This regex probably needs testing in a Windows environment before we can merge. At a bare minimum we need to check that this String will only ever contain "\\"s and not "/"s.

Also is the extra "." in your regex necessary? It seems redundant. If I'm reading it right it's currently matching one or more numbers at the beginning of the string, followed by a single reverse stroke, followed by any character, and then followed by any number of any characters. Why not just ".+"?